homeassistant/components/alexa/capabilities.py

```diff
@@ -715,4 +780,267 @@ def configuration(self):
+
+    def name(self):
+        """Return the Alexa API name of this interface."""
+        return "Alexa.ModeController"
+
+    def properties_supported(self):
+        """Return what properties this entity supports."""
+        return [{"name": "mode"}]
+
+    def properties_proactively_reported(self):
+        """Return True if properties asynchronously reported."""
+        return True
+
+    def properties_retrievable(self):
+        """Return True if properties can be retrieved."""
+
+    def get_property(self, name):
+        """Read and return a property."""
+        if name != "mode":
+            raise UnsupportedProperty(name)
+
+        if self.instance() == fan.ATTR_DIRECTION:
+            return self.entity.attributes.get(fan.ATTR_DIRECTION)
+
+        return None
+
+    def configuration(self):
+        """Return configuration."""
+        configuration = self.serialize_mode_resources()
+        if configuration is not None:
+            return configuration
+        return None
+
+    def capability_resources(self):
+        """Return capabilityResources object."""
+        if self.instance() == fan.ATTR_DIRECTION:
+            return [{"type": Catalog.LABEL_ASSET, "value": Catalog.SETTING_DIRECTION}]
+
+        return None
+
+    def mode_resources(self):
+        """Return modeResources object."""
+        mode_resources = None
+        if self.instance() == fan.ATTR_DIRECTION:
+            mode_resources = {
+                "ordered": False,
+                "resources": [
+                    {
+                        "value": f"{fan.ATTR_DIRECTION}.{fan.DIRECTION_FORWARD}",
+                        "friendly_names": [
+                            {"type": Catalog.LABEL_TEXT, "value": fan.DIRECTION_FORWARD}
+                        ],
+                    },
+                    {
+                        "value": f"{fan.ATTR_DIRECTION}.{fan.DIRECTION_REVERSE}",
+                        "friendly_names": [
+                            {"type": Catalog.LABEL_TEXT, "value": fan.DIRECTION_REVERSE}
+                        ],
+                    },
+                ],
+            }
+
+        return mode_resources
+
+    def serialize_mode_resources(self):
+        """Return ModeResources, friendlyNames serialized for an API response."""
+        mode_resources = []
+        resources = self.mode_resources()
+        ordered = resources["ordered"]
+        for resource in resources["resources"]:
+            mode_value = resource["value"]
+            friendly_names = resource["friendly_names"]
+            result = {
+                "value": mode_value,
+                "modeResources": {
+                    "friendlyNames": self.serialize_friendly_names(friendly_names)
+                },
+            }
+            mode_resources.append(result)
+
+        return {"ordered": ordered, "supportedModes": mode_resources}
+
+
+class AlexaRangeController(AlexaCapibility):
```

Comment:
I would prefer if you try to keep PRs to 1 capability at a time.

Reply:
Also notice a typo in the class name, `AlexaCapability` instead of `AlexaCapibility`

Reply:
These 3 Controllers are related and required the addition of `capabilityResources` in the Alexa responses. Therefore I choose to group them into 1 PR. Having all of them merged together will allow for the additional domain entities attribute/features that can use these controllers to be added quicker as a single PR.

There shouldn't be an issue with the remaining Alexa Controllers only being a single PR.  

That typo lights up my IDE everywhere. I wasn't going to touch it since it was everywhere, and just worked around it.  Happy to refactor, since you mentioned it.